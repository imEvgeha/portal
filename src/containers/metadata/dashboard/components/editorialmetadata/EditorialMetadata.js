import React, { Component } from 'react';
import { Row, Col, Container, TabContent, TabPane, Alert, Tooltip } from 'reactstrap';
import FontAwesome from 'react-fontawesome';
import PropTypes from 'prop-types';
import EditorialMetadataTab from './EditorialMetadataTab';
import EditorialMetadataCreateTab from './EditorialMetadataCreateTab';
import EditorialMetadataEditMode from './EditorialMetadataEditMode';
import { configFields } from '../../../service/ConfigService';
import { connect } from 'react-redux';

const mapStateToProps = state => {
    return {
        configLanguage: state.titleReducer.configData.find(e => e.key === configFields.LANGUAGE)
    };
};

class EditorialMetadata extends Component {
    constructor(props) {
        super(props);
        this.state = {
            tooltipOpen: false
        };
    }

    getLanguageByCode = (code) => {
        if (this.props.configLanguage) {
            const found = this.props.configLanguage.value.find(e => e.languageCode === code);
            if (found) {
                return found.languageName;
            }
        }
        return code;
    };

    toggle = () => {
        this.setState({
            tooltipOpen: !this.state.tooltipOpen
        });
    }

    render() {
        return (
            <Container fluid id="titleContainer" style={{ marginTop: '30px' }}>

                <Row style={{ marginTop: '5px' }}>
                    <Col>
                        <h2>Editorial Metadata</h2>
                    </Col>
                </Row>
                <div className='tab'>
                    {
                        this.props.isEditMode ? (
                            <>
                                <FontAwesome className="tablinks add-local" id="createEditorialMetadata" name="plus-circle" onClick={() => this.props.addEditorialMetadata(this.props.createEditorialTab)} key={this.props.createEditorialTab} size="lg" />
                                <Tooltip placement="top" isOpen={this.state.tooltipOpen} target="createEditorialMetadata" toggle={this.toggle}>
                                    Create Editorial Metadata
                                </Tooltip>
                            </>
                          )
                            : null
                    }
                    {
                        this.props.editorialMetadata && this.props.editorialMetadata.map((item, i) => {
                            return <span className="tablinks" style={{ background: this.props.activeTab === i ? '#000' : '', color: this.props.activeTab === i ? '#FFF' : '' }} key={i} onClick={() => this.props.toggle(i)}><b>{item.locale + ' ' + this.getLanguageByCode(item.language) + ' ' + (item.format ? item.format : '') + ' ' + (item.service ? item.service : '')}</b></span>;
                        })
                    }
                </div>
                <TabContent activeTab={this.props.activeTab}>
                    {
                        this.props.editorialMetadata && this.props.editorialMetadata.length > 0 ?
                            !this.props.isEditMode && this.props.editorialMetadata.map((item, i) => {
                                return (
                                    <TabPane key={i} tabId={i}>
                                        <Row>
                                            <Col>
                                                <EditorialMetadataTab
                                                    titleContentType={this.props.titleContentType}
                                                    getLanguageByCode={this.getLanguageByCode}
                                                    key={i}
                                                    data={item}
                                                />
                                            </Col>
                                        </Row>
                                    </TabPane>
);
                            }) :
                            !this.props.isEditMode ? (
                                <Row>
                                    <Col>
                                        <Alert color="primary">
                                            <FontAwesome name="info" /> <b>No editorial metadata.</b>
                                        </Alert>
                                    </Col>
                                </Row>
                              ) : null
                    }
                    {
                        this.props.isEditMode ? (
                            <>
                                <TabPane tabId={this.props.createEditorialTab}>
                                    <Row>
                                        <Col>
                                            <EditorialMetadataCreateTab
                                                handleAddEditorialCharacterName={this.props.handleAddEditorialCharacterName}
                                                validSubmit={this.props.validSubmit}
                                                areFieldsRequired={this.props.areFieldsRequired}
                                                handleChange={this.props.handleChange}
                                                handleTitleChange={this.props.handleTitleChange}
                                                handleEpisodicChange={this.props.handleEpisodicChange}
                                                editorialMetadataForCreate={this.props.editorialMetadataForCreate}
                                                handleSynopsisChange={this.props.handleSynopsisChange}
                                                handleGenreChange={this.props.handleGenreChange}
                                                handleCategoryChange={this.props.handleCategoryChange}
                                                handleEditorialCastCrewCreate={this.props.handleEditorialCastCrewCreate}
                                                titleContentType={this.props.titleContentType}
                                            />
                                        </Col>
                                    </Row>
                                </TabPane>
                                {
                                    this.props.editorialMetadata && this.props.editorialMetadata.map((item, i) => {
                                        return (
                                            <TabPane key={i} tabId={i}>
                                                <Row>
                                                    <Col>
                                                        <EditorialMetadataEditMode
                                                            handleAddEditorialCharacterNameEdit={this.props.handleAddEditorialCharacterNameEdit}
                                                            titleContentType={this.props.titleContentType}
                                                            validSubmit={this.props.validSubmit}
                                                            handleChange={this.props.handleEditChange}
                                                            handleEditorialCastCrew={this.props.handleEditorialCastCrew}
                                                            handleGenreEditChange={this.props.handleGenreEditChange}
                                                            updatedEditorialMetadata={this.props.updatedEditorialMetadata}
                                                            handleCategoryEditChange={this.props.handleCategoryEditChange}
                                                            key={i}
                                                            data={item}
                                                        />
                                                    </Col>
                                                </Row>
                                            </TabPane>
);
                                    })
                                }
                            </>
                          )
                            : null
                    }
                </TabContent>
            </Container>
        );
    }
}

EditorialMetadata.propTypes = {
    isEditMode: PropTypes.bool.isRequired,
    editorialMetadata: PropTypes.array,
    handleChange: PropTypes.func.isRequired,
    handleTitleChange: PropTypes.func.isRequired,
    handleEpisodicChange: PropTypes.func.isRequired,
    handleSynopsisChange: PropTypes.func.isRequired,
    activeTab: PropTypes.any,
    areFieldsRequired: PropTypes.bool,
    toggle: PropTypes.func,
    addEditorialMetadata: PropTypes.func,
    createEditorialTab: PropTypes.string,
    validSubmit: PropTypes.func.isRequired,
    handleEditChange: PropTypes.func,
    titleContentType: PropTypes.string,
    editorialMetadataForCreate: PropTypes.object,
    updatedEditorialMetadata: PropTypes.array,
    handleGenreChange: PropTypes.func.isRequired,
    handleGenreEditChange: PropTypes.func.isRequired,
    configLanguage: PropTypes.object,
    handleEditorialCastCrewCreate: PropTypes.func,
    handleEditorialCastCrew: PropTypes.func,
    handleAddEditorialCharacterName: PropTypes.func,
    handleAddEditorialCharacterNameEdit: PropTypes.func,
    handleCategoryChange: PropTypes.func.isRequired,
    handleCategoryEditChange: PropTypes.func.isRequired,
};



export default connect(mapStateToProps)(EditorialMetadata);