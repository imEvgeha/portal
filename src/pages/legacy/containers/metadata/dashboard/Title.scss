.lgModal {
    min-width: 90%;
}
