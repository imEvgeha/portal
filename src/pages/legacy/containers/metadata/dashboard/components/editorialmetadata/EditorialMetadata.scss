#editorialMetadataTabs {
    .row {
        padding: 15px;
    }
    .wrap-value {
        word-wrap: break-word;
    }
    #listContainer {
        overflow-y: scroll;
        overflow-x: hidden;
        max-height: 280px;
    }
    .metadata-empty-value {
        color: #999;
    }
    .info-master {
        svg {
            padding-top: 5px;
            margin-right: 5px;
        }
    }
}
#titleContainer {
    .css-lrg2au-singleValue {
        line-height: unset;
    }
}
.tablinks__status-link {
    &--active {
        fill: white;
    }
    width: 15px;
    margin-bottom: 5px;
    margin-right: 5px;
}
