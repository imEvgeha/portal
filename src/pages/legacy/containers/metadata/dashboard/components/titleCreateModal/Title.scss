@import '~@vubiquity-nexus/portal-styles/scss/atlaskit/mixins';
@import '~@vubiquity-nexus/portal-styles/scss/atlaskit/colors';

.lgModal {
    min-width: 90%;
}

.nx-stylish.list-group {
    .alert {
        @include margin(0, 0, 0, 0);
    }
}

.nexus-c-title-create_error-msg {
    color: #bf1650;
}

.nexus-c-title-create_error-msg::before {
    display: inline;
    content: 'âš  ';
}

.p-component.nexus-c-title-create_input {
    @include margin(0, 0, 0, 0);
}

.p-component.nexus-c-title-create_input-number {
    @include margin(0, 0, 0, 0);
    width: 100%;

    input {
        @include margin(0, 0, 0, 0);
    }
}

.nexus-c-title-create_footer-container {
    display: flex;
    align-items: center;
    justify-content: space-between;

    .nexus-c-title-create_footer-buttons-container {
        display: flex;
    }
}

.nexus-c-title-create_input-container {
    @include margin(0, 0, 3, 0);
}

.nexus-c-title-create_checkbox-label {
    margin-left: 8px;
}

.nexus-c-title-create_checkbox-container .row:last-child {
    margin-top: 8px;
}
