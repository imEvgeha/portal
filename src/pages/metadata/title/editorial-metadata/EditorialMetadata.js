import React from 'react';
import PropTypes from 'prop-types';
import NexusPersonsList from '@vubiquity-nexus/portal-ui/lib/elements/nexus-persons-list/NexusPersonsList';
import {CAST_CONFIG, CREW_CONFIG} from '@vubiquity-nexus/portal-ui/lib/elements/nexus-persons-list/constants';
import {getFilteredCastList, getFilteredCrewList} from '@vubiquity-nexus/portal-utils/lib/castCrewUtils';
import {searchPerson} from '../../../avails/right-details/rightDetailsServices';

const EditorialMetadata = ({data, castCrewConfig}) => {
    // TODO: temporary integration of NexusPersonsList here as PoC
    // will be moved after folder structure is implemented
    // no onChange integration done, should be simple
    return (
        <>
            <div>Editorial Metadata</div>
            {data && data.length > 0 && (
                <>
                    Read Only
                    <NexusPersonsList
                        searchPerson={searchPerson}
                        castCrewConfig={castCrewConfig}
                        personsList={getFilteredCastList(data[0].castCrew, false)}
                        hasCharacter
                    />
                    <NexusPersonsList
                        searchPerson={searchPerson}
                        castCrewConfig={castCrewConfig}
                        uiConfig={CREW_CONFIG}
                        personsList={getFilteredCrewList(data[0].castCrew, false)}
                    />
                    Editable
                    <NexusPersonsList
                        searchPerson={searchPerson}
                        castCrewConfig={castCrewConfig}
                        uiConfig={CAST_CONFIG}
                        personsList={getFilteredCastList(data[0].castCrew, false)}
                        isEdit
                        hasCharacter
                    />
                    <NexusPersonsList
                        searchPerson={searchPerson}
                        castCrewConfig={castCrewConfig}
                        uiConfig={CREW_CONFIG}
                        personsList={getFilteredCrewList(data[0].castCrew, false)}
                        isEdit
                    />
                </>
            )}
        </>
    );
};

EditorialMetadata.propTypes = {
    data: PropTypes.array,
    castCrewConfig: PropTypes.object,
};

EditorialMetadata.defaultProps = {
    data: [],
    castCrewConfig: {},
};

export default EditorialMetadata;
