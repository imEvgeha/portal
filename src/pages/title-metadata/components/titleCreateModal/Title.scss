@import '~@vubiquity-nexus/portal-styles/scss/atlaskit/mixins';
@import '~@vubiquity-nexus/portal-styles/scss/atlaskit/colors';

.lgModal {
    min-width: 90%;
}

.nx-stylish.list-group {
    .alert {
        @include margin(0, 0, 0, 0);
    }
}

.nexus-c-title-create_footer-container {
    display: flex;
    align-items: center;
    justify-content: flex-end;

    .nexus-c-title-create_footer-buttons-container {
        display: flex;
    }
}

.nexus-c-title-create_checkbox-label {
    margin-left: 8px;
}

.nexus-c-title-checkbox-wrapper {
    display: flex;
    align-items: center;

    .checkbox-label {
        margin-left: 8px;
    }
}

.nexus-c-title-create_checkbox-wrapper {
    display: flex;
    align-items: baseline;
    padding-left: 18px;
}
