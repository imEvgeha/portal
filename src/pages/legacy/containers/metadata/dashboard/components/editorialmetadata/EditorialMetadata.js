import React, {useEffect, useState} from 'react';
import {Row, Col, Container, TabContent, TabPane, Alert, Tooltip} from 'reactstrap';
import FontAwesome from 'react-fontawesome';
import PropTypes from 'prop-types';
import Button from '@atlaskit/button';
import {connect} from 'react-redux';
import Select from '@atlaskit/select';
import {sortBy} from 'lodash';
import EditorialMetadataTab from './EditorialMetadataTab';
import EditorialMetadataCreateTab from './EditorialMetadataCreateTab';
import EditorialMetadataEditMode from './EditorialMetadataEditMode';
import {configFields} from '../../../service/ConfigService';
import Title from '../../../../../../metadata/title/Title';
import {URL} from '@vubiquity-nexus/portal-utils/lib/Common';
import {NexusDrawer} from '../../../../../../../ui/elements';
import StatusLink from '../../../../../../../assets/status-linked.svg';

const mapStateToProps = state => {
    return {
        configLanguage: state.titleReducer.configData.find(e => e.key === configFields.LANGUAGE),
    };
};

const EditorialMetadata = ({
    isEditMode,
    editorialMetadata,
    handleChange,
    handleAutoDecorateChange,
    handleTitleChange,
    handleEpisodicChange,
    handleSynopsisChange,
    activeTab,
    areFieldsRequired,
    toggle,
    addEditorialMetadata,
    createEditorialTab,
    validSubmit,
    handleEditChange,
    titleContentType,
    editorialMetadataForCreate,
    updatedEditorialMetadata,
    handleGenreChange,
    handleGenreEditChange,
    configLanguage,
    handleEditorialCastCrewCreate,
    handleEditorialCastCrew,
    handleAddEditorialCharacterName,
    handleAddEditorialCharacterNameEdit,
    handleCategoryChange,
    handleCategoryEditChange,
    coreTitleData,
    editorialTitleData,
    cleanField,
    handleDeleteEditorialMetaData,
    handleRegenerateDecoratedMetadata,
    setValidationError,
}) => {
    const [tooltipOpen, setTooltipOpen] = useState(false);
    const [isDrawerOpen, setIsDrawerOpen] = useState(false);
    const [currentFolder, setCurrentFolder] = useState({});
    const [foldersOptions, setFoldersOptions] = useState([]);
    const [foldersChildren, setFoldersChildren] = useState({});

    // When editorialMetadata is received/updated, extract its items and categorize them in folders
    useEffect(() => {
        // Will hold all unique comibnations of locale+language
        const foldersSet = new Set();

        // Extract locale+language from each editorialMetadata item and add it to flodersSet set
        Object.keys(editorialMetadata).forEach(key => {
            const {locale = '', language = ''} = editorialMetadata[key];
            const folderName = `${locale} ${getLanguageByCode(language)}`;

            foldersSet.add(folderName);
        });

        const categorizedEditorialMetadata = {};

        // For each folder, extract its children
        Array.from(foldersSet).forEach(folder => {
            categorizedEditorialMetadata[folder] = editorialMetadata.filter(({locale, language}) => {
                return folder === `${locale} ${getLanguageByCode(language)}`;
            });
        });
        setFoldersChildren(categorizedEditorialMetadata);

        // Create options for react-select
        setFoldersOptions(Array.from(foldersSet).map(folder => ({value: folder, label: folder})));
    }, [editorialMetadata]);
    useEffect(() => setCurrentFolder(foldersOptions[0]), [foldersOptions]);

    const titleHasMaster = editorialMetadata.find(e => e['hasGeneratedChildren']);
    const isNexusTitle = coreTitleData.id && coreTitleData.id.startsWith('titl');

    const {value: currentFolderName = ''} = currentFolder || {};

    const getLanguageByCode = code => {
        if (configLanguage) {
            const found = configLanguage.value.find(e => e.languageCode === code);
            if (found) {
                return found.languageName;
            }
        }
        return code;
    };

    const masterFirstFoldersChildren = sortBy(foldersChildren[currentFolderName], ['hasGeneratedChildren']).reverse();
    return (
        <Container fluid id="titleContainer" style={{marginTop: '30px'}}>
            <NexusDrawer isOpen={isDrawerOpen} onClose={() => setIsDrawerOpen(false)} width="extended" position="right">
                <Title coreTitleData={coreTitleData} editorialTitleData={editorialTitleData} />
            </NexusDrawer>
            <Row style={{marginTop: '5px'}}>
                <Col>
                    <h2>Editorial Metadata</h2>
                </Col>
                <Col style={{display: 'flex', justifyContent: 'flex-end'}}>
                    {URL.isLocalOrDev() && <Button onClick={() => setIsDrawerOpen(true)}>Open drawer</Button>}
                    {isNexusTitle &&
                        titleHasMaster &&
                        !isEditMode &&
                        currentFolder &&
                        currentFolderName === 'US English' && (
                            <Button
                                appearance="primary"
                                style={{marginLeft: '15px'}}
                                onClick={handleRegenerateDecoratedMetadata}
                            >
                                Regenerate Auto-Decorated Metadata
                            </Button>
                        )}
                </Col>
            </Row>
            <Row>
                <div style={{width: '200px', margin: '5px 0 15px 15px'}}>
                    <Select
                        options={foldersOptions}
                        defaultValue={foldersOptions[0]}
                        value={currentFolder || foldersOptions[0]}
                        onChange={folder => {
                            toggle(0);
                            setCurrentFolder(folder);
                        }}
                    />
                </div>
            </Row>
            <div className="tab">
                {isEditMode ? (
                    <>
                        <FontAwesome
                            className="tablinks add-local"
                            id="createEditorialMetadata"
                            name="plus-circle"
                            onClick={() => addEditorialMetadata(createEditorialTab)}
                            key={createEditorialTab}
                            size="lg"
                        />
                        <Tooltip
                            placement="top"
                            isOpen={tooltipOpen}
                            target="createEditorialMetadata"
                            toggle={() => setTooltipOpen(!tooltipOpen)}
                        >
                            Create Editorial Metadata
                        </Tooltip>
                    </>
                ) : null}
                {masterFirstFoldersChildren &&
                    masterFirstFoldersChildren.map((item, index) => {
                        const isDecorated = !!item['parentEmetId'];
                        const isMaster = item['hasGeneratedChildren'];
                        return (
                            <span
                                className="tablinks"
                                style={{
                                    background: activeTab === index ? '#000' : '',
                                    color: activeTab === index ? '#FFF' : '',
                                }}
                                key={index}
                                onClick={() => toggle(index)}
                            >
                                {(isMaster || isDecorated) && (
                                    <StatusLink
                                        className={`tablinks__status-link ${
                                            activeTab === index ? 'tablinks__status-link--active' : ''
                                        }`}
                                    />
                                )}
                                <b>
                                    {`${item.locale} ${getLanguageByCode(item.language)} ${item.format || ''} ${
                                        item.service || ''
                                    }`}
                                </b>
                            </span>
                        );
                    })}
            </div>
            <TabContent activeTab={activeTab}>
                {masterFirstFoldersChildren ? (
                    !isEditMode &&
                    masterFirstFoldersChildren.map((item, i) => {
                        return (
                            <TabPane key={i} tabId={i}>
                                <Row>
                                    <Col>
                                        <EditorialMetadataTab
                                            titleContentType={titleContentType}
                                            getLanguageByCode={getLanguageByCode}
                                            key={i}
                                            data={item}
                                        />
                                    </Col>
                                </Row>
                            </TabPane>
                        );
                    })
                ) : !isEditMode ? (
                    <Row>
                        <Col>
                            <Alert color="primary">
                                <FontAwesome name="info" /> <b>No editorial metadata.</b>
                            </Alert>
                        </Col>
                    </Row>
                ) : null}
                {isEditMode ? (
                    <>
                        <TabPane tabId={createEditorialTab}>
                            <Row>
                                <Col>
                                    <EditorialMetadataCreateTab
                                        handleAddEditorialCharacterName={handleAddEditorialCharacterName}
                                        validSubmit={validSubmit}
                                        areFieldsRequired={areFieldsRequired}
                                        handleChange={handleChange}
                                        handleAutoDecorateChange={handleAutoDecorateChange}
                                        handleTitleChange={handleTitleChange}
                                        handleEpisodicChange={handleEpisodicChange}
                                        editorialMetadataForCreate={editorialMetadataForCreate}
                                        handleSynopsisChange={handleSynopsisChange}
                                        handleGenreChange={handleGenreChange}
                                        handleCategoryChange={handleCategoryChange}
                                        handleEditorialCastCrewCreate={handleEditorialCastCrewCreate}
                                        titleContentType={titleContentType}
                                        cleanField={cleanField}
                                        titleHasMaster={titleHasMaster}
                                        setValidationError={setValidationError}
                                    />
                                </Col>
                            </Row>
                        </TabPane>
                        {masterFirstFoldersChildren &&
                            masterFirstFoldersChildren.map((item, i) => {
                                return (
                                    <TabPane key={i} tabId={i}>
                                        <Row>
                                            <Col>
                                                <EditorialMetadataEditMode
                                                    handleAddEditorialCharacterNameEdit={
                                                        handleAddEditorialCharacterNameEdit
                                                    }
                                                    titleContentType={titleContentType}
                                                    validSubmit={validSubmit}
                                                    handleChange={handleEditChange}
                                                    handleEditorialCastCrew={handleEditorialCastCrew}
                                                    handleGenreEditChange={handleGenreEditChange}
                                                    updatedEditorialMetadata={updatedEditorialMetadata}
                                                    handleCategoryEditChange={handleCategoryEditChange}
                                                    handleDelete={handleDeleteEditorialMetaData}
                                                    key={i}
                                                    data={item}
                                                    setValidationError={setValidationError}
                                                />
                                            </Col>
                                        </Row>
                                    </TabPane>
                                );
                            })}
                    </>
                ) : null}
            </TabContent>
        </Container>
    );
};

EditorialMetadata.propTypes = {
    isEditMode: PropTypes.bool.isRequired,
    editorialMetadata: PropTypes.array,
    handleChange: PropTypes.func.isRequired,
    handleAutoDecorateChange: PropTypes.func.isRequired,
    handleTitleChange: PropTypes.func.isRequired,
    handleEpisodicChange: PropTypes.func.isRequired,
    handleSynopsisChange: PropTypes.func.isRequired,
    cleanField: PropTypes.func.isRequired,
    activeTab: PropTypes.any,
    areFieldsRequired: PropTypes.bool,
    toggle: PropTypes.func,
    addEditorialMetadata: PropTypes.func,
    createEditorialTab: PropTypes.string,
    validSubmit: PropTypes.func.isRequired,
    handleEditChange: PropTypes.func,
    titleContentType: PropTypes.string,
    editorialMetadataForCreate: PropTypes.object,
    updatedEditorialMetadata: PropTypes.array,
    handleGenreChange: PropTypes.func.isRequired,
    handleGenreEditChange: PropTypes.func.isRequired,
    configLanguage: PropTypes.object,
    handleEditorialCastCrewCreate: PropTypes.func,
    handleEditorialCastCrew: PropTypes.func,
    handleAddEditorialCharacterName: PropTypes.func,
    handleAddEditorialCharacterNameEdit: PropTypes.func,
    coreTitleData: PropTypes.object,
    editorialTitleData: PropTypes.array,
    handleCategoryChange: PropTypes.func.isRequired,
    handleCategoryEditChange: PropTypes.func.isRequired,
};

EditorialMetadata.defaultProps = {
    editorialMetadata: [],
    areFieldsRequired: false,
    toggle: () => null,
    addEditorialMetadata: () => null,
    createEditorialTab: '',
    handleEditChange: () => null,
    titleContentType: '',
    editorialMetadataForCreate: {},
    updatedEditorialMetadata: [],
    configLanguage: {value: []},
    handleEditorialCastCrew: () => null,
    handleEditorialCastCrewCreate: () => null,
    handleAddEditorialCharacterName: () => null,
    handleAddEditorialCharacterNameEdit: () => null,
    coreTitleData: {},
    editorialTitleData: [],
};

export default connect(mapStateToProps)(EditorialMetadata);
