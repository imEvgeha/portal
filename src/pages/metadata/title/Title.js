import React, {useState} from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import TitleHeader from './title-header/TitleHeader';
import EditorialMetadata from './editorial-metadata/EditorialMetadata';

import {
    METADATA_TITLE_TITLE_SECTION,
    METADATA_TITLE_CAST_N_CREW_SECTION,
    METADATA_TITLE_RATINGS_SECTION,
    METADATA_TITLE_EXTERNAL_IDS_SECTION,
    METADATA_TITLE_EDITORIAL_SECTION,
    METADATA_TITLE_TERRITORIAL_SECTION,
    METADATA_TITLE_RIGHTS_SECTION,
    METADATA_TITLE_SYNC_LOG_SECTION,
    METADATA_TITLE_SECTIONS,
} from './constants';
import './Title.scss';

const Title = ({
    coreTitleData,
    editorialTitleData,
}) => {
    const [currentSection, setCurrentSection] = useState(METADATA_TITLE_EDITORIAL_SECTION);
    const {title, releaseYear, type} = coreTitleData || {};

    const renderSection = section => {
        switch (section) {
            case METADATA_TITLE_TITLE_SECTION:
                return null;
            case METADATA_TITLE_CAST_N_CREW_SECTION:
                return null;
            case METADATA_TITLE_RATINGS_SECTION:
                return null;
            case METADATA_TITLE_EXTERNAL_IDS_SECTION:
                return null;
            case METADATA_TITLE_EDITORIAL_SECTION:
                return (
                    <EditorialMetadata
                        data={editorialTitleData}
                    />
                );
            case METADATA_TITLE_TERRITORIAL_SECTION:
                return null;
            case METADATA_TITLE_RIGHTS_SECTION:
                return null;
            case METADATA_TITLE_SYNC_LOG_SECTION:
                return null;
            default:
                return null;
        }
    };

    return (
        <div className="nexus-c-metadata-title">
            <TitleHeader
                title={title}
                releaseYear={releaseYear}
                type={type}
            />
            <div className="nexus-c-metadata-title__sections-menu">
                {METADATA_TITLE_SECTIONS.map((section, index) => (
                    <div
                        className={classNames(
                            'nexus-c-metadata-title__section-tab',
                            (section === currentSection) && 'nexus-c-metadata-title__section-tab--is-active'
                        )}
                        key={index}
                        onClick={() => setCurrentSection(section)}
                    >
                        {section}
                    </div>
                ))}
            </div>
            <div className="nexus-c-metadata-title__section">
                {renderSection(currentSection)}
            </div>
        </div>
    );
};

Title.propTypes = {
    coreTitleData: PropTypes.object.isRequired,
    editorialTitleData: PropTypes.array.isRequired,
};

Title.defaultProps = {};

export default Title;
