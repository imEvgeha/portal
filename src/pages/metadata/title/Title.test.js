import React from 'react';
import {shallow} from 'enzyme';

import Title from './Title';
import EditorialMetadata from './editorial-metadata/EditorialMetadata';
import {
    METADATA_TITLE_SECTIONS,
<<<<<<< HEAD
    METADATA_TITLE_TITLE_SECTION,
=======
>>>>>>> a3bbc79c4929ec53116d0110b3433600821b33c2
    METADATA_TITLE_EDITORIAL_SECTION,
} from './constants';

describe('Title', () => {
    let wrapper = null;
    const sectionTabClass = '.nexus-c-metadata-title__section-tab';

    beforeEach(() => {
        wrapper = shallow(
            <Title
                coreTitleData={{
                    title: 'AtlasKit 101',
                    type: 'Comedy',
                    releaseYear: '2020',
                }}
            />
        );
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render tabs', () => {
        expect(wrapper.find(sectionTabClass)).toHaveLength(METADATA_TITLE_SECTIONS.length);
    });

    it('should render section when respective tab is clicked', () => {
        const EMetTab = wrapper.findWhere(
            node => node.hasClass('nexus-c-metadata-title__section-tab') && node.text() === METADATA_TITLE_EDITORIAL_SECTION
        );
        EMetTab.simulate('click');
        expect(wrapper.contains(<EditorialMetadata />)).toBeTruthy();
    });
});
