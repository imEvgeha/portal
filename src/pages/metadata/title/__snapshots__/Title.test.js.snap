// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should match snapshot 1`] = `
<div
  className="nexus-c-metadata-title"
>
  <TitleHeader
    artwork=""
    releaseYear="2020"
    title="AtlasKit 101"
    type="Comedy"
  />
  <div
    className="nexus-c-metadata-title__sections-menu"
  >
    <div
      className="nexus-c-metadata-title__section-tab"
      key="0"
      onClick={[Function]}
    >
      Title
    </div>
    <div
      className="nexus-c-metadata-title__section-tab"
      key="1"
      onClick={[Function]}
    >
      Cast & Crew
    </div>
    <div
      className="nexus-c-metadata-title__section-tab"
      key="2"
      onClick={[Function]}
    >
      Ratings
    </div>
    <div
      className="nexus-c-metadata-title__section-tab"
      key="3"
      onClick={[Function]}
    >
      External IDs
    </div>
    <div
      className="nexus-c-metadata-title__section-tab nexus-c-metadata-title__section-tab--is-active"
      key="4"
      onClick={[Function]}
    >
      Editorial
    </div>
    <div
      className="nexus-c-metadata-title__section-tab"
      key="5"
      onClick={[Function]}
    >
      Territorial
    </div>
    <div
      className="nexus-c-metadata-title__section-tab"
      key="6"
      onClick={[Function]}
    >
      Rights
    </div>
    <div
      className="nexus-c-metadata-title__section-tab"
      key="7"
      onClick={[Function]}
    >
      Sync Log
    </div>
  </div>
  <div
    className="nexus-c-metadata-title__section"
  >
    <EditorialMetadata
      data={Array []}
    />
  </div>
</div>
`;
